sklearn/tests/test_discriminant_analysis.py

```diff
@@ -176,8 +175,12 @@ def test_lda_explained_variance_ratio():
     clf_lda_svd = LinearDiscriminantAnalysis(solver="svd")
     clf_lda_svd.fit(X, y)
     assert_almost_equal(clf_lda_svd.explained_variance_ratio_.sum(), 1.0, 3)
+
+    # NOTE: clf_lda_eigen.explained_variance_ratio_ is not of n_components
+    # length. Make it the same length as clf_lda_svd.explained_variance_ratio_
+    # before comparison.
     assert_array_almost_equal(clf_lda_svd.explained_variance_ratio_,
-                              clf_lda_eigen.explained_variance_ratio_)
+            clf_lda_eigen.explained_variance_ratio_[:3])
```

Comment:
If `n_components=None`, then by default, the svd and eigen
LinearDiscriminantAnalysis classes will return all of the eigenvalues. It
just turns out that the eigen solver yields more eigenvalues than the SVD
solver.

On Tue, Dec 15, 2015 at 5:29 PM, JPFrancoia <notifications@github.com>
wrote:

> In sklearn/tests/test_discriminant_analysis.py
> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_scikit-2Dlearn_scikit-2Dlearn_pull_6027-23discussion-5Fr47712339&d=BQMCaQ&c=8hUWFZcy2Z-Za5rBPlktOQ&r=IgVP_sCIQHhgRgxQoo7gufqSQOvlFIckTGgYIpXBa_4&m=AY2fz0E96atHTtLqc2ACdb4QVEt00CicRvw0s637MTI&s=Zu0YSrErMT2_MEUBieaV6Oxmv6w7p4Qp3yVNkt1lUCI&e=>
> :
>
> >      assert_array_almost_equal(clf_lda_svd.explained_variance_ratio_,
> > -                              clf_lda_eigen.explained_variance_ratio_)
> > +            clf_lda_eigen.explained_variance_ratio_[:3])
>
> I mean, why is the sdv array of length 3 ? We don't specify n_components
> when we initialize the classifiers.
>
> —
> Reply to this email directly or view it on GitHub
> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_scikit-2Dlearn_scikit-2Dlearn_pull_6027_files-23r47712339&d=BQMCaQ&c=8hUWFZcy2Z-Za5rBPlktOQ&r=IgVP_sCIQHhgRgxQoo7gufqSQOvlFIckTGgYIpXBa_4&m=AY2fz0E96atHTtLqc2ACdb4QVEt00CicRvw0s637MTI&s=GQ8PKCIJDMszV5ezTabub8KKiTGyOE7H6dg4rMfFm08&e=>
> .
>



-- 



*Henry Lin, Research AssistantM.S. Computer ScienceUniversity of Illinois
at Urbana-Champaign 2016847-769-8729 | halin2@illinois.edu
<halin2@illinois.edu>*